karlstad-universitet-harvard.csl

```diff
@@ -0,0 +1,359 @@
+  </info>
+  <locale xml:lang="sv">
+    <terms>
+      <term name="page" form="short">
+        <single>s.</single>
+        <multiple>ss.</multiple>
+      </term>
+      <term name="number-of-pages" form="short">
+        <single>s.</single>
+        <multiple>ss.</multiple>
+      </term>
+    </terms>
+  </locale>
+  <macro name="author">
+    <choose>
+      <if type="motion_picture broadcast" match="none">
+        <names variable="author">
+          <name and="symbol" delimiter-precedes-last="never" initialize-with=". " name-as-sort-order="all"/>
+          <label form="short" prefix=" (" suffix=")."/>
+          <et-al font-style="italic"/>
+          <substitute>
+            <names variable="editor"/>
+            <names variable="translator"/>
+            <text macro="anon"/>
+          </substitute>
+        </names>
+      </if>
+      <else>
+        <choose>
+          <if variable="container-title">
+            <text variable="container-title" font-style="italic"/>
+          </if>
+          <else>
+            <text variable="title" font-style="italic"/>
+          </else>
+        </choose>
+      </else>
+    </choose>
+    <choose>
+      <if type="article-newspaper entry-dictionary entry-encyclopedia" match="any">
+        <choose>
+          <if variable="author editor" match="none">
+            <text variable="container-title"/>
+          </if>
+        </choose>
+      </if>
+    </choose>
+  </macro>
+  <macro name="author-short">
+    <choose>
+      <if type="motion_picture broadcast" match="none">
+        <names variable="author">
+          <name form="short" delimiter=" " and="symbol" delimiter-precedes-last="never" initialize-with=". " name-as-sort-order="all"/>
+          <substitute>
+            <names variable="editor"/>
+            <names variable="translator"/>
+            <text macro="anon"/>
+          </substitute>
+        </names>
+      </if>
+      <else>
+        <choose>
+          <if variable="container-title">
+            <text variable="container-title" font-style="italic"/>
+          </if>
+          <else>
+            <text variable="title" font-style="italic"/>
+          </else>
+        </choose>
+      </else>
+    </choose>
+    <choose>
+      <if type="article-newspaper entry-dictionary entry-encyclopedia" match="any">
+        <choose>
+          <if variable="author editor" match="none">
+            <text variable="container-title"/>
+          </if>
+        </choose>
+      </if>
+    </choose>
+  </macro>
+  <macro name="anon">
+    <choose>
+      <if type="motion_picture broadcast article-newspaper entry-dictionary entry-encyclopedia" match="none">
+        <text term="anonymous" form="short" text-case="capitalize-first"/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="access">
+    <choose>
+      <if variable="URL" type="webpage post-weblog">
```

Comment:
This won't work. If you want the access macro to print if a URL is present _and_ the item is a webpage _or_ a blog post you'll need two consecutive choose-ifs. (This might work in Zotero but it's wrong and won't work in other tools -- see e.g. the preview here.

Reply:
I changed it to 
`  </macro>
  <macro name="access">
    <choose>
      <if type="webpage">
        <group>
          <text value="Tillgängligt: "/>
          <text variable="URL"/>
        </group>
        <group prefix=" [" suffix="]" delimiter=" ">
          <text value="hämtad" text-case="capitalize-first"/>
          <date variable="accessed">
            <date-part name="year"/>
            <date-part name="month" form="numeric" prefix="-"/>
            <date-part name="day" prefix="-"/>
          </date>
        </group>
      </if>
    </choose>
  </macro>`